src/clr-icons/shapes/core-shapes.ts

```diff
@@ -304,54 +303,84 @@ export const ClrShapeStepForward2 = clrIconSVG(
-  'warning-standard': ClrShapeWarningStandard,
-  'help-info': ClrShapeHelpInfo,
-  bars: ClrShapeBars,
-  user: ClrShapeUser,
+export const CoreShapes = {
+  'angle-double': ClrShapeAngleDouble,
   angle: ClrShapeAngle,
-  folder: ClrShapeFolder,
-  'folder-open': ClrShapeFolderOpen,
+  bars: ClrShapeBars,
   bell: ClrShapeBell,
-  image: ClrShapeImage,
+  calendar: ClrShapeCalendar,
+  'check-circle': ClrShapeCheckCircle,
+  check: ClrShapeCheck,
   cloud: ClrShapeCloud,
+  cog: ClrShapeCog,
   'ellipsis-horizontal': ClrShapeEllipsisHorizontal,
   'ellipsis-vertical': ClrShapeEllipsisVertical,
-  'filter-grid': ClrShapeFilterGrid,
-  'filter-grid-circle': ClrShapeFilterGridCircle,
-  'vm-bug': ClrShapeVmBug,
-  search: ClrShapeSearch,
-  'view-columns': ClrShapeViewColumns,
-  'angle-double': ClrShapeAngleDouble,
-  calendar: ClrShapeCalendar,
+  'error-standard': ClrShapeErrorStandard,
   event: ClrShapeEvent,
-  eye: ClrShapeEye,
+  'exclamation-circle': ClrShapeExclamationCircle,
+  'exclamation-triangle': ClrShapeExclamationTriangle,
   'eye-hide': ClrShapeEyeHide,
+  eye: ClrShapeEye,
+  'filter-grid-circle': ClrShapeFilterGridCircle,
+  'filter-grid': ClrShapeFilterGrid,
+  'folder-open': ClrShapeFolderOpen,
+  folder: ClrShapeFolder,
+  'help-info': ClrShapeHelpInfo,
+  home: ClrShapeHome,
+  image: ClrShapeImage,
+  'info-circle': ClrShapeInfoCircle,
+  'info-standard': ClrShapeInfoStandard,
+  search: ClrShapeSearch,
   'step-forward-2': ClrShapeStepForward2,
+  'success-standard': ClrShapeSuccessStandard,
+  times: ClrShapeTimes,
+  'unknown-status': ClrShapeUnknownStatus,
+  user: ClrShapeUser,
+  'view-columns': ClrShapeViewColumns,
+  'vm-bug': ClrShapeVmBug,
+  'warning-standard': ClrShapeWarningStandard,
+  /** Alias for: angle */ get caret(): string {
+    return this['angle'];
+  },
+  /** Alias for: angle-double */ get collapse(): string {
+    return this['angle-double'];
+  },
+  /** Alias for: bars */ get menu(): string {
+    return this['bars'];
+  },
+  /** Alias for: bell */ get notification(): string {
+    return this['bell'];
+  },
+  /** Alias for: check */ get success(): string {
+    return this['check'];
+  },
+  /** Alias for: cog */ get settings(): string {
+    return this['cog'];
+  },
+  /** Alias for: exclamation-circle */ get error(): string {
+    return this['exclamation-circle'];
+  },
+  /** Alias for: exclamation-triangle */ get warning(): string {
+    return this['exclamation-triangle'];
+  },
+  /** Alias for: folder */ get directory(): string {
+    return this['folder'];
+  },
+  /** Alias for: home */ get house(): string {
+    return this['home'];
+  },
+  /** Alias for: info-circle */ get info(): string {
+    return this['info-circle'];
+  },
+  /** Alias for: times */ get close(): string {
+    return this['times'];
+  },
+  /** Alias for: user */ get avatar(): string {
+    return this['user'];
+  },
 };
 
-Object.defineProperty(CoreShapes, 'house', descriptorConfig(CoreShapes.home));
```

Comment:
Due to concerns about regressions, please revert the `Object.defineProperty` calls in all the files and remove the getters. We have to test these changes extensively by building the icons and importing them into several different types of applications (both Angular CLI, standalone) and that work will be done as part of https://github.com/vmware/clarity/issues/2599.

Reply:
This PR can wait until all fixes are done, as I said in a previous comment. Rolling back to Object.defineProperty not only makes the purpose of this PR meaningless (because typing will not work as expected, due to lack of type inference with Object.defineProperty on TypeScript), but also after the build steps are fixed, this will have to be "rolled back again" to getters, so is better option to just wait until the build is fixed.

Reply:
I agree with @gnomeontherun's comment. Also, I don't see this PR becoming meaningless if do reverts as he suggested. The value we were are gaining from these type enforcement on `getters` is not worth it compare to the potential breaks we might introduce to the library. The UglifyJS is one of the deeply nested dependencies so we don't have much control over it. We are especially wary of this change at this moment because we are in the process of delivering Clarity 2.0. 

We could introduce this kind of type enforcement changes when we start tackling #2599. By then, we will run extensive integration, unit, and other tests.

Reply:
This PR makes it so when you import any set of icons, you have auto-completition of the icons in that set, and errors if you try to use an icon that doesn't exist. TypeScript as I said before can't infer anything from `Object.defineProperty` so all alias will be marked as "errors" and will not get auto-completition.

I'll wait until you solve the issues you have with your build (I didn't find any running the build, but I take your word on that), and once it works with getters and setters (kind of a core functionality nowadays), I'll update this PR.

Reply:
The build runs just fine, but consuming the build in a new project and testing it is not something we have built into our primary testing pipeline right now.